### ISSUE TITLE
AMP by Wordpress.com with Understap Nav Menu Not Showing

### ISSUE THREAD

**Issue by zachatkinson** [understrap/understrap#693]
## Issue is about..
  - [ x ] a Feature Request
  - [ x ] a Question
  - [ x ] I need help

## Please tell us about your environment:

  - **WordPress Version:** 4.9.6
  - **OS Version:** MacOS [10.13.4]
  - **Browser:** [ all ]
  - **Language:** [PHP | HTML | S/CSS | Javascript]

---

# Issue Overview for Bugreports
Recently I've been launching a new client site using a child of Understrap, my preferred WP theme. One of the steps I'm taking on this site is to enable AMP pages. I've installed the AMP plugin by Automattic, as well as Yoast SEO Glue to help config it. It would seem as though no menu shows on AMP pages with Understrap themes.  Not sure if anyone else has had success implementing AMP with Understrap.  Worst case scenario I can modify the walker and write my own custom code, but I was hoping perhaps I had just overlooked something simple

## Steps to reproduce?

1. Install AMP (by Automattic/Wordpress.com)
2. Set up AMP configuration (enable for posts & pages)
3. Install Yoast SEO Glue (Optional, but I use Yoast SEO Professional so this makes things a little easier) 
4. View amp post on your site (append /amp/ to your post address, or ?amp to a page address)
5. Note lack of menu

## *Who* is the bug affecting?
End users
## *What* is affected by this bug?
Navigation Menu
## *When* does this occur?
When AMP pages are loaded
## *Where* on the platform does it happen?
In the header
## *How* do we replicate the issue?
Follow Steps above

## Expected behavior
It would be nice to see understrap write nav by default (or through a function option) with proper amp-sidebar conversion for amp pages.

### Other Comments

I've pooled some resources to help figure out what is happening here. Not sure if they will be of use to anyone else.

AMP Plugin - https://en-ca.wordpress.org/plugins/amp/
Yoast SEO Glue - https://wordpress.org/plugins/glue-for-yoast-seo-amp/
Similar Question asked at AMP Github - https://github.com/Automattic/amp-wp/issues/792

**Comment by pattonwebz** [comment]
I wasn't aware there were specific issues with the walker implementation when used with AMP. Nobody has reported it to me in the walker repo :confused: 

If you know what needs changed or why it's not working then I can put the fix in the upstream repo and make a PR to merge it here.

**Comment by zachatkinson** [comment]
Wow, thanks for your incredibly fast response.  I'll be digging in heavy today to find out whats wrong.  If you want a live example check out 

https://manhattan-construction.com/about-manhattan-construction/

https://manhattan-construction.com/about-manhattan-construction/amp

I'm pretty sure if I follow that "similar question asked at AMP git" I can get it going, but I didn't find that until late last night so I've yet to take a stab.  Any advice appreciated - and hey - you are providing a free and EXCELLENT product, I'm certainly not suggesting the walker doesn't rock! It could very well be me using it wrong. 

Thanks again for your fast reply!

-Zach

**Comment by pattonwebz** [comment]
As for the fast response: push notifications are a wonderful thing ;)

I'm just taking a look at your linked resources now. I didn't know that AMP menus worked in this way as I do not have much experience working with it. From the sounds of things it might actually be best to use a conditional in a filter to check for AMP request and then overload the walker with one from a different custom written class.

I don't think the walker currently in use would be able to be filtered easily to modify the markup and classes enough to make it work as an AMP sideloading nav - which is why I suggested a different custom written one.

I'm happy to help write the custom walker if that winds up being the best option here :)

**Comment by zachatkinson** [comment]
This is my first experience using AMP too - so being unfamiliar with their best practices probably doesn't help matters :). I agree the best course of action would be to check for AMP request and decide whether to load the sidebar or just behave as intended as a normal menu in the header.  

### Working With AMP
One of the difficulties I have found while working on this AMP project is how little info is actually available.  For a while, it was suggested not to have a menu as AMP was meant for strictly single-page articles, but now they have updated things to give us greater flexibility.  I will tell you this - out of all the optimizations I have done for the site this has been the biggest pain.  Even simple display CSS has to be re-written.  Looks like Google is going to continue to push it though, so I bit the bullet and decided to dig in.  

I'm going to see if I can't make this bad boy work today and then I'll share with you my (probably inferior) code for reference if you would like, a new walker allowing this out of the box would be super awesome!  

I've attached a few more links below of examples and docs just for quick reference 😅 

### AMP Sidebar Reference Material
https://ampbyexample.com/components/amp-sidebar/preview/
https://ampbyexample.com/components/amp-sidebar/
https://www.ampproject.org/docs/reference/components/amp-sidebar

### Food For Thought on Adding Support to Understrap
There is, of course, a question to be raised of whether this is "out of spec" for Understrap since AMP is not enabled by default and Understrap works as intended on a proper WP install. I would argue as AMP is pushed more by Google it would be a boon for us to be prepared and have a built-in solution to handle menu issues, even if it means the Understrap package becomes slightly larger.

**Comment by Thomas-A-Reinert** [comment]
@pattonwebz thanks for jumping in into that issue! Really.

Never met that problem, but i´m not really AMP-optimizing my sites (yet)

Do you need other dev-environments to evalue? If that´s the case please let me know.

**Comment by Hoodlum-Z** [comment]
@zachatkinson I agree that implementing AMP is the biggest pain ever, but like you, I decided it was worthwhile.  I'm glad I have done it because I've learned so much this last fortnight. It's enabled me to delete about 6 plugins, plus my site is fully AMP.  Only thing I can't resolve yet is my menu, I've included the AMP navbar, but it means I have a text menu and a hamburger menu instead of just the one menu.
